src/map.rs

```diff
@@ -79,22 +77,116 @@ where
 }
 
 /// Message is the response struct from the [`Mapper::map`] .
+#[derive(Debug, PartialEq)]
 pub struct Message {
     /// Keys are a collection of strings which will be passed on to the next vertex as is. It can
     /// be an empty collection.
-    pub keys: Vec<String>,
+    pub keys: Option<Vec<String>>,
     /// Value is the value passed to the next vertex.
     pub value: Vec<u8>,
     /// Tags are used for [conditional forwarding](https://numaflow.numaproj.io/user-guide/reference/conditional-forwarding/).
-    pub tags: Vec<String>,
+    pub tags: Option<Vec<String>>,
+}
+/// Represents a message that can be modified and forwarded.
+impl Message {
+    /// Creates a new message with the specified value.
+    ///
+    /// This constructor initializes the message with no keys, tags, or specific event time.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - A vector of bytes representing the message's payload.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use numaflow::map::Message;
+    /// let message = Message::new(vec![1, 2, 3, 4]);
+    /// ```
+    pub fn new(value:Vec<u8>) -> Self {
+        Self{
+            value,
+            keys:None,
+            tags:None
+        }
+    }
+    /// Marks the message to be dropped by adding a special "DROP" tag.
+    ///
+    /// This method guarantees that the tags vector is initialized if it was previously `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use numaflow::map::Message;
+    /// let dropped_message = Message::new(vec![1, 2, 3]).message_to_drop();
+    /// ```
+    pub fn message_to_drop(mut self) -> Self {
+        if self.tags.is_none(){
+            self.tags=Some(Vec::new());
+        }
+        self.tags.as_mut().unwrap().push(DROP.parse().unwrap());
+        self
+    }
```

Comment:
@yhl25, @shubhamdixit863; isn't it better to have a standalone `message_to_drop`? what is the point of `new().message_to_drop()`, esp. since `new` takes a mandatory arg?

Reply:
message_to_drop() should be stand alone!